pallets/parachain-staking/src/lib.rs

```diff
@@ -986,14 +976,12 @@ pub mod pallet {
 					if remaining.is_zero() {
 						<DelegatorState<T>>::remove(&bond.owner);
 					} else {
-						if let Some(request) = delegator.requests.requests.remove(&candidate) {
-							delegator.requests.less_total =
-								delegator.requests.less_total.saturating_sub(request.amount);
-							if matches!(request.action, DelegationChange::Revoke) {
-								delegator.requests.revocations_count =
-									delegator.requests.revocations_count.saturating_sub(1u32);
-							}
-						}
+						Self::delegation_remove_request_with_state(
+							&candidate,
+							&bond.owner,
+							&mut delegator,
+						)
+						.ok(); // ignore DNE error
```

Comment:
Does this mean we will silently ignare a DNE error? Is this a change in behavior? Elaborating on this comment would help.